pgrx-tests/src/tests/proptests.rs

```diff
@@ -0,0 +1,77 @@
+use crate::proptest::PgTestRunner;
+use pgrx::prelude::*;
+use proptest::prelude::*;
+use proptest::strategy::Strategy;
+
+#[pg_extern]
+pub fn nop_date(date: Date) -> Date {
+    date
+}
+
+#[cfg(any(test, feature = "pg_test"))]
+#[pgrx::pg_schema]
+mod tests {
+    use super::*;
+    use crate as pgrx_tests;
+
+    // Property tests consist of 1:
+    /// Hypothesis: We can pass random dates directly into Postgres functions and get them back.
+    #[pg_test]
+    pub fn date_spi_roundtrip() {
+        // 2. Constructing the Postgres-adapted test runner
+        let mut proptest = PgTestRunner::default();
+        // 3. A strategy for creating and refining values
+        let strat = prop::num::i32::ANY.prop_map(|i| {
+            Date::try_from(i).unwrap_or(unsafe {
+                Date::from_pg_epoch_days(i32::clamp(i, -2451545, 2147483494 - 2451546))
+            })
```

Comment:
This should be extracted into a function on the Date type.

Reply:
This is important for the correctness of the changes in this test, so I can't skimp on it before landing this.